features/FxA-83-sign-in-confirmation-email/README.md

```diff
@@ -0,0 +1,160 @@
+to maliciously access users’ Firefox Accounts.
+In such cases,
+attackers would have full access
+to the user’s Sync data,
+including passwords, history, bookmarks and
+the ability to force-install malicious add-ons.
+
+## Outcomes
+
+If an attack occurs,
+we would like fewer accounts to be compromised
+without negatively impacting
+the number of signed-in users
+or our overall FxA engagement rate.
+
+## Hypothesis
+
+If we introduce an additional email confirmation step
+for Sync sign-ins,
+fewer accounts will be compromised.
+We will know this to be true if,
+in the event of an attack,
+the number of successful sign-ins remains constant
+and the count of 104 (unverified account) errors
+from protected endpoints
+increases in line with any wider increase in traffic.
+
+## Assumptions
+
+* The `/account/login` endpoint
+  is solely responsible for initiating
+  the confirmation process.
+
+* The following endpoints should be protected:
+  * `/certificate/sign`
+  * `/account/keys`
+  * `/account/delete`
+  * `/password/change/start`
+
+* In most cases,
+  attackers do not have access
+  to our users' email accounts.
+  Users that do have a compromised email account
+  are out of scope for this feature.
+
+## Constraints
+
+* In light of the recent attack,
+  the solution should be quick to implement.
+
+* The solution must work across all clients,
+  without patches landing in client code.
+
+* The solution must not affect existing sessions
+  from the user's point of view.
+
+* The solution must work from
+  regular Sync sign-in,
+  `/force_auth` and
+  when signing-in to Sync from `/signup`.
+
+## Proposed solution
+
+Add a notion of “verified”
+to sessionTokens and keyFetchTokens.
+Tokens that are unverified have reduced powers.
+Token verification is achieved
+by following a link sent by email.
+
+## User stories
+
+* As a Sync user,
+  when signing in to my Firefox Account,
+  I want my identity confirmed
+  via my email account.
+
+* As a user of an OAuth relier,
+  when signing in to my Firefox Account,
+  I do not want an extra confirmation step.
+
+## Work breakdown
+
+### fxa-content-server
+
+- [ ] Add strings to `strings.js`
+  to get localization done quicker.
+- [ ] Implement "confirm your email" screen.
+- [ ] Implement confirmation landing screen.
+- [ ] Add handling for `verified` and `challenge` fields
+  in `/account/login` response.
+
+### fxa-auth-server
+
+- [ ] Add method to `mailer.js`
+  for sending confirmation email.
+- [ ] Modify `/account/login`
+  to send verification status to db,
+  initiate verification email
+  and set `challenge` on response.
+- [ ] In `/account/delete`,
+  fail with 102 (unverified user) error
+  if sessionToken is not verified.
+- [ ] In `/certificate/sign`,
+  fail with 102 (unverified user) error
+  if sessionToken is not verified.
+- [ ] In `/account/keys`,
+  fail with 102 (unverified user) error
+  if keyFetchToken is not verified.
+- [ ] In `/password/change/start`,
```

Comment:
> `/password/change/start` does not currently require a `sessionToken`, that may need to change?

That is what I meant by `require a sessionToken` in the very next line. Maybe it isn't clear, care to suggest better text?

> Can we easily return an `unverified` keyFetchToken similar to an unverified sessionToken, meaning this endpoint could also initiate the "extra verification required" step?

We definitely can do that, but I'm not sure it is something we should focus on right now. Given that the aim is to have this deployed by the end of the week, focusing on sign-ins in the first cut seems prudent.